core/grant/service.go

```diff
@@ -257,3 +270,176 @@ func (s *Service) expiredInActiveUserAccess(ctx context.Context, timeLimiter cha
 		}
 	}
 }
+
+type ImportAccessCriteria struct {
+	ProviderID   string `validate:"required"`
+	ResourceIDs  []string
+	ResouceTypes []string
+	ResourceURNs []string
+}
+
+func (s *Service) ImportAccess(ctx context.Context, criteria ImportAccessCriteria) ([]*domain.Grant, error) {
+	p, err := s.providerService.GetByID(ctx, criteria.ProviderID)
+	if err != nil {
+		return nil, fmt.Errorf("getting provider details: %w", err)
+	}
+
+	listResourcesFilter := domain.ListResourcesFilter{
+		ProviderType: p.Type,
+		ProviderURN:  p.URN,
+	}
+	if criteria.ResourceIDs != nil {
+		listResourcesFilter.IDs = criteria.ResourceIDs
+	} else {
+		listResourcesFilter.ResourceTypes = criteria.ResouceTypes
+		listResourcesFilter.ResourceURNs = criteria.ResourceURNs
+	}
+	resources, err := s.resourceService.Find(ctx, listResourcesFilter)
```

Comment:
Is it possible to just have a List resource that return resources that has populated provider struct?
So we are just doing 1 DB call to get all resources.
I assume number of resources per provider is not that many.

Reply:
based on the way we are using it right now, this service is called in `POST /providers/:provider_id/import-access` API, so IMO it's still fair to fetch the provider details first, and then the resources